schema/encoding/jsonschema.go

```diff
@@ -0,0 +1,110 @@
+package jsonschema
+
+import (
+	"fmt"
+	"github.com/rs/rest-layer/schema"
+	"strconv"
+	"strings"
+)
+
+// JSONSchemaPropertyBase common Embedded Values
+type JSONSchemaPropertyBase struct {
+	Title string `json:"title"`
+	Type  string `json:"type"`
+}
+
+// JSONSchema represents a standard JSONSchema
+type JSONSchema struct {
+	JSONSchemaPropertyBase
+	Properties map[string]map[string]interface{} `json:"properties"`
+}
+
+// BoundariesToJSONSchema render boundaries to JSON
+func BoundariesToJSONSchema(b *schema.Boundaries) string {
+	if b == nil {
+		return ""
+	}
+	return fmt.Sprintf("%q: %s, %q: %s",
+		"minimum", strconv.FormatFloat(b.Min, 'E', -1, 64),
+		"maximum", strconv.FormatFloat(b.Max, 'E', -1, 64))
+}
+
+// ValidatorToJSONSchema takes a validator and renders to JSON
+func ValidatorToJSONSchema(v schema.FieldValidator) string {
+	if v == nil {
+		return ""
+	}
+	var result string
+	switch t := v.(type) {
+	case *schema.String:
+		result += fmt.Sprintf("%q: %q, ", "type", "string")
+		if t.Regexp != "" {
+			result += fmt.Sprintf("%q: %q, ", "pattern", t.Regexp)
+		}
+		if len(t.Allowed) > 0 {
+			var allowed []string
+			for _, value := range t.Allowed {
+				allowed = append(allowed, fmt.Sprintf("%q", value))
+			}
+			result += fmt.Sprintf("%q: [ %s ],", "oneOf", strings.Join(allowed, ","))
```

Comment:
Sorry.. keep type as it is, switch key "oneOf" with key "enum" that's all... read the spec wrong there (reviewing on my phone)